<launch>
  <arg name="web_video_port" default="8088"/>
  <arg name="rosbridge_port" default="9091"/>
  <arg name="marker_length" default="0.03"/>  <arg name="aruco_dict" default="DICT_4X4_50"/>
  <arg name="main_marker_id" default="0"/>

  <group ns="front_cam">
    <node pkg="front_cam_aruco" type="robust_aruco_node.py" name="robust_aruco_node" output="screen">
      <param name="image_topic" value="/usb_camera/image_raw"/>
      <param name="camera_info_topic" value="/usb_camera/camera_info"/>
      <param name="show_gui" value="false"/>
      <param name="log_interval" value="3.0"/>
      </node>

    <!-- RL Target Visualizer -->
    <node pkg="front_cam_aruco" type="rl_target_visualizer.py" name="rl_target_visualizer" output="screen">
      <param name="camera_frame" value="front_cam/usb_cam_link"/>
      <param name="robot_base_frame" value="base_link"/>
      <param name="image_topic" value="/front_cam/usb_cam/robust_debug"/>
      <param name="target_topic" value="/front_cam/rl_target"/>
    </node>

    <node pkg="web_video_server" type="web_video_server" name="web_video_server" output="screen">
      <param name="port" value="$(arg web_video_port)"/>
    </node>

    <node pkg="rosbridge_server" type="rosbridge_websocket" name="rosbridge_websocket" output="screen">
      <param name="port" value="$(arg rosbridge_port)"/>
    </node>
  </group>

  <node pkg="tf2_ros" type="static_transform_publisher" name="front_cam_static_tf"
        args="0.08 0 0 -1.57079632679 0 -1.57079632679 base_link front_cam/usb_cam_link" />
</launch>
